modules/porous_flow/src/dirackernels/PorousFlowTimeLimitedConstantPointSource.C

```diff
@@ -0,0 +1,64 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowTimeLimitedConstantPointSource.h"
+
+template<>
+InputParameters validParams<PorousFlowTimeLimitedConstantPointSource>()
+{
+  InputParameters params = validParams<DiracKernel>();
+  params.addRequiredParam<Real>("mass_flux", "The mass flux at this point in kg/s (positive is flux in, negative is flux out)");
+  params.addRequiredParam<std::vector<Real> >("point", "The x,y,z coordinates of the point");
+  params.addParam<Real>("end_time", 1.0e30, "The time at which the source will end");
+  return params;
+}
+
+PorousFlowTimeLimitedConstantPointSource::PorousFlowTimeLimitedConstantPointSource(const InputParameters & parameters) :
+    DiracKernel(parameters),
+    _mass_flux(getParam<Real>("mass_flux")),
+    _point_param(getParam<std::vector<Real> >("point")),
+    _end_time(getParam<Real>("end_time"))
+{
+  _p(0) = _point_param[0];
+
+  if (_point_param.size() > 1)
+    _p(1) = _point_param[1];
+  if (_point_param.size() > 2)
+      _p(2) = _point_param[2];
```

Comment:
see my previous comment about `<Point>`. I believe it would make all this logic unnecessary.